src/memory.js

```diff
@@ -2,17 +2,43 @@ class MemoryGame {
   constructor(cards) {
     this.cards = cards;
     // add the rest of the class properties here
+
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
   }
 
   shuffleCards() {
-    // ... write your code here
+
+    if (!this.cards) {
+      return undefined;
+    }
+    for (let i = this.cards.length -1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [this.cards[i], this.cards[j]] = [this.cards[j], this.cards[i]];
+    }
+    return this.cards;
   }
 
   checkIfPair(card1, card2) {
     // ... write your code here
+    this.pairsClicked ++;
+
+    if (card1 === card2) {
+      this.pairsGuessed++;
+      return true;
+    } else {
+      return false;
+    }
   }
 
   checkIfFinished() {
     // ... write your code here
+
+    if (this.pairsGuessed === this.cards.length / 2) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
Esto de aqui, fijate que estas devolviendo un true y false, entonces no sera mejor devolver directamente la condicion del if, acuerdate quie eso resuelve precisamente en un booleano!! :D

```suggestion
    return this.pairsGuessed === this.cards.length / 2;
```

Reply:
es verdad! gracias :) :)